core/sys/cc.h

```diff
@@ -111,6 +111,19 @@
 #endif /* CC_CONF_INLINE */
 
 /**
+ * Auto-detect the toolchain's support for weak symbols
+ */
+#if __STDC_VERSION__ >= 199901L
+#define CC_PRAGMA(x) _Pragma(#x)
+#endif
+
+#if __GNUC__
+#define CC_WEAK_ALIAS(alias, ref) __attribute__((__weak__, __alias__(#ref)))
+#elif __ICCARM__ || __ICCSTM8__
+#define CC_WEAK_ALIAS(alias, ref) CC_PRAGMA(weak alias=ref)
+#endif
```

Comment:
This was a failed attempt to do what was discussed above. I've reverted to the previous version of the pull, the entire discussion should be visible now